include/flags.h

```diff
@@ -23,6 +24,14 @@ using argument_map =
 struct parser {
   parser(const int argc, char** argv) {
     for (int i = 1; i < argc; ++i) {
+      // If this token is "--", skip it and stop parsing
+      if (strcmp(argv[i], "--") == 0) {
```

Comment:
nit: consider string_view(argv[i]) == "--" here and we can avoid the cstring header and it's a bit easier to use than strcmp

Reply:
Since argv[i] is a `const char*` we couldn't do a direct comparison as that would compare the pointers themselves. We could create a `std::string_view` from argv[i] then compare against that:
```c++
if (std::string_view(argv[i]) == "--") {
```
My opinion is that looks messier, although it would allow removing the cstring include

Reply:
Going to merge this, but this could always be changed in the future